container/memory_store.go

```diff
@@ -0,0 +1,91 @@
+package container
+
+import "sync"
+
+// memoryStore implements a Store in memory.
+type memoryStore struct {
+	s map[string]*Container
+	sync.Mutex
```

Comment:
I'm not sure if we want to change this, but I'd be okay with doing it. My only concern is this scenario:

1. I terminate the daemon, the store holds a RLock.
2. Someone else starts a new container, adding it to the store, the store holds a RWLock.
3. The daemon finishes and we lose a container in the termination chain because of the RLock.